app/src/main/java/de/danoeh/antennapod/fragment/EpisodesFragment.java

```diff
@@ -4,125 +4,240 @@
-
-        ViewPager2 viewPager = rootView.findViewById(R.id.viewpager);
-        viewPager.setAdapter(new EpisodesPagerAdapter(this));
-        viewPager.setOffscreenPageLimit(2);
-        super.setupPagedToolbar(toolbar, viewPager);
-
-        // Give the TabLayout the ViewPager
-        tabLayout = rootView.findViewById(R.id.sliding_tabs);
-
-        new TabLayoutMediator(tabLayout, viewPager, (tab, position) -> {
-            switch (position) {
-                case POS_NEW_EPISODES:
-                    tab.setText(R.string.new_episodes_label);
-                    break;
-                case POS_ALL_EPISODES:
-                    tab.setText(R.string.all_episodes_short_label);
-                    break;
-                case POS_FAV_EPISODES:
-                    tab.setText(R.string.favorite_episodes_label);
-                    break;
-                default:
-                    break;
-            }
-        }).attach();
 
-        // restore our last position
-        SharedPreferences prefs = getActivity().getSharedPreferences(TAG, Context.MODE_PRIVATE);
-        int lastPosition = prefs.getInt(PREF_LAST_TAB_POSITION, 0);
-        viewPager.setCurrentItem(lastPosition, false);
+        //setUpQuickFilter();
+
+        setSwipeActions(TAG);
+
+        return  rootView;
+    }
 
-        return rootView;
+    @Override
+    public void onStart() {
+        super.onStart();
+        SharedPreferences prefs = requireActivity().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        updateFloatingFilterButton(prefs.getString(PREF_POSITION, QUICKFILTER_ALL));
     }
 
     @Override
     public void onPause() {
         super.onPause();
-        // save our tab selection
-        SharedPreferences prefs = getActivity().getSharedPreferences(TAG, Context.MODE_PRIVATE);
-        SharedPreferences.Editor editor = prefs.edit();
-        editor.putInt(PREF_LAST_TAB_POSITION, tabLayout.getSelectedTabPosition());
-        editor.apply();
+    }
+
+    public String getPrefFilter() {
+        SharedPreferences prefs = requireActivity().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        return prefs.getString(PREF_FILTER, "");
+    }
+
+    private void updateFloatingFilterButton(String id) {
+        String newFilter;
+        int menuitemId;
+        switch (id) {
+            default:
+            case EpisodesListFragment.QUICKFILTER_ALL:
+                newFilter = getPrefFilter();
+                menuitemId = R.id.filter_all;
+                break;
+            case EpisodesListFragment.QUICKFILTER_UNPLAYED:
+                newFilter = FeedItemFilter.UNPLAYED;
+                menuitemId = R.id.filter_unplayed;
+                break;
+            case EpisodesListFragment.QUICKFILTER_DOWNLOADED:
+                newFilter = FeedItemFilter.DOWNLOADED;
+                menuitemId = R.id.filter_downloaded;
+                break;
+            case EpisodesListFragment.QUICKFILTER_FAV:
```

Comment:
Didn't you remove the quick filters? I think they should only be added to the code when we have decided how to deal with them.

Reply:
Yea sorry forgot that part